README.md

```diff
@@ -1,277 +1,899 @@
+of swaps in the future.
 
-| Undesirable Property | Workaround "Solution" | Issues |
-| :--: | :--: | :--: |
-| Impermanent Loss | Yield Farming & Concentrated Liquidity | Medium - Long term unsustainability |
-| Incomplete or No Delegation Control | <ul><li>Asset pool fractionalization</li><li>Indirect Delegation via Governance Tokens</li></ul> | <ul><li>Unfair distribution of Governance tokens </li><li>Unavoidable centralization of stake (Major issue for Ouroboros)</li></ul>
-| Scaling/Availability Bottlenecks | Batchers, Execution/Routing Engines, and/or other middlemen | Middlemen can take advantage of their position between users and the protocol. Even if MEV is mitigated, more users --> more execution demand --> possible centralization of middlemen if the batchers are not permissionless |
+:exclamation: While the spec shows one-way swap datums and redeemers having the same names as the
+two-way swap datums and redeemers *they are distinct data types*. Imagine they are in separate name
+spaces.
 
+### One-Sway Swaps
 
-### Programmable Swaps
-First proposed by Axo in their original [whitepaper](https://www.axo.trade/whitepaper.pdf), programmable swaps are a more promising design choice than liquidity pools. Swaps are simply UTxOs that may be consumed if and only if the resulting TX outputs satisfy the input script's logic. Swaps can be fragmented across many user-controlled addresses, so delegation control is maintained. Users then execute swaps from each other's addresses. Since each swap is atomic and explicitly defined, in aggregate they are the optimal expression of (intra)market sentiment. This design pattern scales naturally, since there must be at *least* as many swap addresses as there are users. 
+One-way swaps use one universal validator script for all swap pairs, and one universal minting
+policy for all one-way beacons. There are two beacons for one-way swaps: the trading pair beacon and
+the offer beacon.
 
-### The Cardano-Swaps Protocol
-Cardano-Swaps takes inspiration from Axo's programmable swaps design, adds delegation control as a foundational feature, and, through the use of Beacon Tokens, removes the need for specialized indexers. The only remaining bottleneck is the querying capacity of existing off-chain APIs, such as Blockfrost or Koios. (This is not a limitation for users with powerful enough hardware, as they can run their own API database).
+The offer beacon asset name is `sha2_256( offer_policy_id ++ offer_asset_name )`.
 
-The Cardano-Swaps Protocol is broadly comprised of two steps:
+The trading pair beacon asset name is 
+`sha2_256( offer_id ++ offer_name ++ ask_id ++ ask_name )`. However, in the case where one
+of the assets is ADA (which is just the empty bytestring), the ADA policy id is set to "00" instead.
+Without this changed, there would be no difference between the offer beacon and trading pair beacon
+when ADA is part of a swap.
```

Comment:
Maybe an example?  Would ADA->DJED look like this?:
sha2_256( "00" ++ "ADA" ++ "1234" ++ "DJED")

Reply:
No, it would be: `sha2_256( "00" ++ "" ++ "1234" ++ "DJED")`
For one-way swaps, the other direction would be: `sha2_256( "1234" ++ "DJED" ++ "00" ++ "")`

I can add an example to the readme.